### ISSUE TITLE
Spiky Loss Curves

### ISSUE THREAD

**Issue by haok1402** [cmu-flame/FLAME-MoE#3]
> I was running some reproduction runs but I see my loss curves are spiky. I wonder if this is about my training environment or just the training configurations. Could you please share your loss figures? I would greatly appreaciate a point of comparison.
> 
> <img width="1004" height="588" alt="Image" src="https://github.com/user-attachments/assets/2ebca0bb-9b47-4c8d-8e5f-9cfc5717ded9" /> 

 _Originally posted by @luukkonenr in [#2](https://github.com/cmu-flame/FLAME-MoE/issues/2#issuecomment-3310921246)_

**Comment by haok1402** [comment]
Different loss curves, including those for scaling law studies and final releases, are fully available inside our public [WandB](https://wandb.ai/haok/flame-moe). These include the standard cross-entropy loss for next-token prediction, load balancing loss for mitigating routing collapses, and z-loss for penalizing large-magnitude router logits.

The exact loss curve figure with respect to the 290m scale can be found [here](https://wandb.ai/haok/flame-moe/groups/flame-moe-290m/workspace). It wasn't as spiky as yours. @luukkonenr Could you share more configurations of yours, such as the dataset it is being trained on, and other hyper-parameters?

<img width="2040" height="825" alt="Image" src="https://github.com/user-attachments/assets/77d02758-3dac-4af5-912f-bcb2d602b263" />

**Comment by haok1402** [comment]
@luukkonenr I'm closing the issue now due to inactivity. Feel free to follow up if you get any other related questions!
